students/models.py

```diff
@@ -39,24 +36,21 @@ class Student(Person):
                                on_delete=models.SET_NULL)
     invited = models.IntegerField(default=0, null=True)
 
-    @classmethod
-    def generate_instances(cls, count):
-        faker = Faker()
-        for _ in range(count):
-            st = cls(
-                first_name=faker.first_name(),
-                last_name=faker.last_name(),
-                email=faker.email(),
-                birthdate=faker.date_time_between(start_date="-30y", end_date="-18y"),
-            )
-            st.save()
+    @receiver(post_save, sender=User)
```

Comment:
Move it from class part to simple functions. I should be on models level:
class Student
    some logic here

def signal():
    signal logic

Reply:
Because if that signal is not working. Run debug and try to check why this error is raised. 
I tried to reset password.

Internal Server Error: /reset/MQ/set-password/
Traceback (most recent call last):
  File "D:\Hillel\lesson_14\homeworks\Homeworks\venv\lib\site-packages\django\core\handlers\exception.py", line 47, in inner
    response = get_response(request)
  File "D:\Hillel\lesson_14\homeworks\Homeworks\venv\lib\site-packages\django\core\handlers\base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "D:\Hillel\lesson_14\homeworks\Homeworks\venv\lib\site-packages\django\views\generic\base.py", line 70, in view
    return self.dispatch(request, *args, **kwargs)
  File "D:\Hillel\lesson_14\homeworks\Homeworks\venv\lib\site-packages\django\utils\decorators.py", line 43, in _wrapper
    return bound_method(*args, **kwargs)
  File "D:\Hillel\lesson_14\homeworks\Homeworks\venv\lib\site-packages\django\views\decorators\debug.py", line 89, in sensitive_post_parameters_wrapper
    return view(request, *args, **kwargs)
  File "D:\Hillel\lesson_14\homeworks\Homeworks\venv\lib\site-packages\django\utils\decorators.py", line 43, in _wrapper
    return bound_method(*args, **kwargs)
  File "D:\Hillel\lesson_14\homeworks\Homeworks\venv\lib\site-packages\django\views\decorators\cache.py", line 44, in _wrapped_view_func
    response = view_func(request, *args, **kwargs)
  File "D:\Hillel\lesson_14\homeworks\Homeworks\venv\lib\site-packages\django\contrib\auth\views.py", line 272, in dispatch
    return super().dispatch(*args, **kwargs)
  File "D:\Hillel\lesson_14\homeworks\Homeworks\venv\lib\site-packages\django\views\generic\base.py", line 98, in dispatch
    return handler(request, *args, **kwargs)
  File "D:\Hillel\lesson_14\homeworks\Homeworks\venv\lib\site-packages\django\views\generic\edit.py", line 142, in post
    return self.form_valid(form)
  File "D:\Hillel\lesson_14\homeworks\Homeworks\venv\lib\site-packages\django\contrib\auth\views.py", line 301, in form_valid
    user = form.save()
  File "D:\Hillel\lesson_14\homeworks\Homeworks\venv\lib\site-packages\django\contrib\auth\forms.py", line 356, in save
    self.user.save()
  File "D:\Hillel\lesson_14\homeworks\Homeworks\venv\lib\site-packages\django\contrib\auth\base_user.py", line 67, in save
    super().save(*args, **kwargs)
  File "D:\Hillel\lesson_14\homeworks\Homeworks\venv\lib\site-packages\django\db\models\base.py", line 726, in save
    self.save_base(using=using, force_insert=force_insert,
  File "D:\Hillel\lesson_14\homeworks\Homeworks\venv\lib\site-packages\django\db\models\base.py", line 774, in save_base
    post_save.send(
  File "D:\Hillel\lesson_14\homeworks\Homeworks\venv\lib\site-packages\django\dispatch\dispatcher.py", line 180, in send
    return [
  File "D:\Hillel\lesson_14\homeworks\Homeworks\venv\lib\site-packages\django\dispatch\dispatcher.py", line 181, in <listcomp>
    (receiver, receiver(signal=self, sender=sender, **named))
  File "D:\Hillel\lesson_14\homeworks\Homeworks\students\models.py", line 46, in save_user_profile
    instance.profile.save()
  File "D:\Hillel\lesson_14\homeworks\Homeworks\venv\lib\site-packages\django\db\models\fields\related_descriptors.py", line 421, in __get__
    raise self.RelatedObjectDoesNotExist(
django.contrib.auth.models.User.profile.RelatedObjectDoesNotExist: User has no profile.